scoringrules/backend/numpy.py

```diff
@@ -224,6 +224,20 @@ def where(self, condition: "NDArray", x1: "NDArray", x2: "NDArray") -> "NDArray"
     def size(self, x: "NDArray") -> int:
         return x.size
 
+    def inv(self, x: "NDArray") -> "NDArray":
+        return np.linalg.inv(x)
+
+    def cov(self, x: "NDArray") -> "NDArray":
+        # TODO: This deserves some axis parameter probably?
+        if len(x.shape) == 2:
+            return np.cov(x)
+        else:
+            centered = x - x.mean(axis=-2, keepdims=True)
+            return (centered.transpose(0, 2, 1) @ centered) / (x.shape[-2] - 1)
```

Comment:
centered.transpose(0, 2, 1) assumes the array is 3D. Is there a way to generalise this to calculate the covariance matrix for higher dimensions?

Reply:
If we assume that we always have the square matrix in the last two dimensions (like numpy does on the batched version of e.g. `np.linalg.inv()`), something like `centered.swapaxes(-1, -2)` should do the trick. Is that convention fine for you @sallen12?